integration/spark/src/main/scala/org/apache/spark/sql/hive/CarbonRelation.scala

```diff
@@ -207,7 +207,7 @@ case class CarbonRelation(
                   null != validSeg.getLoadMetadataDetails.getIndexSize) {
                 size = size + validSeg.getLoadMetadataDetails.getDataSize.toLong +
                        validSeg.getLoadMetadataDetails.getIndexSize.toLong
-              } else {
+              } else if (!carbonTable.isHivePartitionTable) {
```

Comment:
why add this check?

Reply:
Here, it aims collect the datasize of segment path. but the format of segment path generated is "Fart/Part0/Segment_0".
For partition table. will throw out FileNotFound exception.

Reply:
better to find the root cause and fix it. LoadMetadataDetail should have data/index size beside of old store.

Reply:
modified